library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -592,23 +592,33 @@ protected void replay(long windowId)
                 pendingFiles.remove(recoveryEntry.file);
               }
               inputStream = retryFailedFile(new FailedFile(recoveryEntry.file, recoveryEntry.startOffset));
+
+              while (--skipCount >= 0) {
+                readEntity();
+              }
               while (offset < recoveryEntry.endOffset) {
                 T line = readEntity();
                 offset++;
                 emit(line);
               }
+              if (recoveryEntry.fileClosed) {
+                closeFile(inputStream);
+              }
             } else {
               while (offset < recoveryEntry.endOffset) {
                 T line = readEntity();
                 offset++;
                 emit(line);
               }
+              if (recoveryEntry.fileClosed) {
+                closeFile(inputStream);
+              }
             }
           }
         }
       }
     } catch (IOException e) {
-      throw new RuntimeException("replay", e);
+      failureHandling(e);
```

Comment:
Looks like, with failure in reading from a file it skips all other recovery entries in the window and continues with next window (if within max failure count). If there are other files to be read within this window, they would be skipped.